# Workflow: Pathway Enrichment I

## Overview

This repository is part of the [Pathway Commons Guide](http://pathwaycommons.github.io/guide/). It contains code relevant to the workflow for [Pathway Enrichment -  I](http://pathwaycommons.github.io/guide/workflows/pathway_enrichment_i/index/). Packaged within a [Docker container](https://hub.docker.com/r/jvwong/) is an in-browser version of [RStudio](https://hub.docker.com/r/rocker/rstudio/) containing the necessary [Bioconductor](https://www.bioconductor.org/) dependencies to generate the files necessary to perform the enrichment workflow.

---

## Getting started

### Software requirements

- [Docker](https://docs.docker.com/engine/installation/): version > 1.12
- [Docker Compose](https://docs.docker.com/compose/): version > 1.9

### Descriptions

You should see the following directory structure.

```shell
|--- pathway_enrichment_i
|   |
|   |--- output
|   |
|   |--- scripts
|   |    |
|   |    |--- get_data.R
|   |    |--- process_data.txt
|   |    |--- visualize.txt
|   |
|   |--- docker-compose.yml
...
```

- `docker-compose.yml`: Instructs Docker as to what images to pull from the DockerHub and runs the RStudio
- `/output`: Files generated in the R session will be saved here
- `/scripts`
  - [`get_data.R`](http://pathwaycommons.github.io/guide/workflows/pathway_enrichment_gdc/get_data/): We source high-grade serous ovarian cancer (HGS-OvCa) RNA sequencing (RNA-seq) data generated by [The Cancer Genome Atlas (TCGA)](https://cancergenome.nih.gov/) effort and housed at the [Genonomic Data Commons (GDC)](https://gdc.cancer.gov/).
  - [`process_data.R`](http://pathwaycommons.github.io/guide/workflows/pathway_enrichment_gdc/process_data/): The data are analyzed for differential expression between HGS-OvCa subtypes and the genes are ranked accordingly. The ranks are the input for an [enrichment analysis performed using GSEA](http://pathwaycommons.github.io/guide/workflows/pathway_enrichment_gdc/identify_pathways/)
  - [`visualize.R`](http://pathwaycommons.github.io/guide/workflows/pathway_enrichment_gdc/visualize/): The output of GSEA is visualized using an [Enrichment Map](http://apps.cytoscape.org/apps/enrichmentmap). Data formatted as part of this step are optional inputs for an Enrichment Map visualization.

### Start

1. Build, create, start, and attach to containers for the rstudio service.

  Assuming you have Docker and Docker Compose, open up a terminal and navigate to the directory `pathway_enrichment_i`.

    ```shell
    $ docker-compose up -d
    ```

  > Note: A Docker image will be pulled from the DockerHub. It will be run and link to the `scripts` and `output` directories inside `/home/TCGA`. This means that changes written to `output` by the R scripts will be available, even after the container exits.

3. Point your browser to [`localhost:8787`](http://localhost:8787)

4. Use RStudio!

  The username and password are both `rstudio`. In RStudio, the directories will be avilable at `/home/TCGA` (click the `...` button in the Files tab).

## Troubleshooting

1. Running Docker through a [virtual machine](https://docs.docker.com/machine/get-started/)

  You can find the IP address by typing `echo $(docker-machine ip machinename)` where 'machinename' is the name of your virtual machine (e.g. 'default').

2. Port conflicts

  Feel free to modify the 'ports' attribute inside the 'rstudio' service within  `docker-compose.yml` if you get a port conflict on your machine.

<hr/>
